app/models/user.rb

```diff
@@ -1,6 +1,22 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  has_many :reviews
+  has_many :cakes # !
+  has_many :cakes, through: :orders, dependent: :destroy # not sure
```

Comment:
Dependent destroy should be removed here I think, and one more association we need ,above this line, for this line to work is has_many :orders